crates/goose-server/src/routes/config_management.rs

```diff
@@ -734,6 +746,41 @@ pub async fn update_custom_provider(
     Ok(Json(format!("Updated custom provider: {}", id)))
 }
 
+#[utoipa::path(
+    post,
+    path = "/config/check_provider",
+    request_body = CheckProviderRequest,
+)]
+pub async fn check_provider(
+    Json(CheckProviderRequest { provider }): Json<CheckProviderRequest>,
+) -> Result<(), (StatusCode, String)> {
+    create_with_default_model(&provider)
+        .await
+        .map_err(|err| (StatusCode::BAD_REQUEST, err.to_string()))?;
+    Ok(())
+}
+
+#[utoipa::path(
+    post,
+    path = "/config/set_provider",
+    request_body = SetProviderRequest,
+)]
+pub async fn set_provider(
```

Comment:
should we call this one set_config_provider and the other set_agent_provider for easier reading?

Reply:
yeah, good idea